BLSQ/iaso
# Workflow changes management

Allow user to manage workflow changes  (CRUD)
See [Whimsical](https://whimsical.com/web-interface-5JjLtfS6my7DwLqqoaaTTT) for mockups

Changes are containing  mapping between target (a form you have to select) and source (the reference form of a workflow)
By adding a mapping, you'll decide if a field of the target should affect one of the field of the reference form. 

Related JIRA tickets : WC2-113, WC2-114, WC2-129, WC2-130, WC2-132, WC2-122


## Self proof reading checklist


## Changes

- bask-end api to CRUD changes for a specific workflow version
- front-end adaptation to make it possible
- fix search in workflow versions list
- fix deep copy of a workflow version

## How to test

Go to a list of workflow versions  for a beneficiary type, try to search a version by name
Try to make a duplication of a version
Go to the detail of a draft workflow version.
Play with the changes keeping in mind that:
- Source and target should be of the same type
- You cannot have multiple mappings on the same source or target
- target form can be used only once in the changes


## Print screen / video

https://user-images.githubusercontent.com/12494624/214072335-f64a8c23-3ac5-4fba-9748-36af7c282538.mov


## Notes

Test will be done on separate PR's (WC2-132, WC2-133)
Sorry for the huge PR
